File: 6. Weekly Pickup/final6.py
import pandas as pd 
import datetime 
import matplotlib.pyplot as plt 
import numpy as np

all_date = []
#=============================================================================================================#
df = pd.read_csv("uber_nyc_enriched.csv")
for row in df.iterrows():
    row[1]["pickup_dt"] = datetime.datetime.strptime(row[1]["pickup_dt"], "%Y-%m-%d %H:%M:%S")
    all_date.append(row[1]["pickup_dt"].strftime('%A'))

df["pickup_dt"] = all_date
# new_df = pd.pivot_table(df, index=["pickup_dt"], values=['pickups'], aggfunc=np.mean)
# df.plot.box()
# print(new_df)

query_table = df.query('pickup_dt == ["Sunday"] and borough == ["Manhattan"]')
sunday = query_table.pickups

query_table = df.query('pickup_dt == ["Monday"] and borough == ["Manhattan"]')
monday = query_table.pickups

query_table = df.query('pickup_dt == ["Tuesday"] and borough == ["Manhattan"]')
tuesday = query_table.pickups

query_table = df.query('pickup_dt == ["Wednesday"] and borough == ["Manhattan"]')
wednesday = query_table.pickups

query_table = df.query('pickup_dt == ["Thursday"] and borough == ["Manhattan"]')
thursday = query_table.pickups

query_table = df.query('pickup_dt == ["Friday"] and borough == ["Manhattan"]')
friday = query_table.pickups

query_table = df.query('pickup_dt == ["Saturday"] and borough == ["Manhattan"]')
saturday = query_table.pickups

data_to_plot = np.array([sunday, monday, tuesday, wednesday, thursday, friday, saturday])

plt.boxplot(data_to_plot)
plt.xlabel('Days')
plt.ylabel('Pickups')
plt.title('Box Plot of Pickups in Manhattan')
plt.xticks(range(1, 8, 1), ('Sun', 'Mon', 'Tue', 'Wed', 'Thurs', 'Fri', 'Sat'))
plt.show()

File: 6. Weekly Pickup/final6.R
library(dplyr)
library(tidyr)
library(ggplot2)

filename <- "uber_nyc_enriched.csv"
df <- read.csv(filename, header = TRUE)

df$pickup_dt <- weekdays(as.POSIXct(df$pickup_dt), abbreviate = FALSE)

query_table = subset(df, df$borough=='Manhattan')
query_table$pickup_dt <- factor(query_table$pickup_dt, c("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"))
# query_table[order(query_table$pickup_dt)]

# pivot <- df %>%
#   select(pickup_dt, pickups) %>%
#   group_by(pickup_dt) %>%
#   summarise(total_pickups = sum(pickups))

ggplot(query_table, aes(x=pickup_dt, y=pickups))+geom_boxplot()
